fns-hl7-pipeline/fn-structural-validator/src/main/java/structruralvalidator/Function.scala

```diff
@@ -4,88 +4,89 @@ import com.microsoft.azure.functions.ExecutionContext
 import com.microsoft.azure.functions.annotation.EventHubTrigger
 import com.microsoft.azure.functions.annotation.FunctionName
 
-import java.util.Optional;
-
-import scala.util.{Try, Failure, Success}
-import scala.concurrent.duration._
-import java.util.concurrent.TimeUnit
-
-import scala.concurrent._
-
+import scala.util.{Failure, Success, Try}
 import cdc.xlr.structurevalidator._
+import com.google.gson.{Gson, GsonBuilder, JsonElement, JsonParser}
 
+import java.util
+import scala.collection.BitSet.empty.to
 
 /**
  * Azure Functions with Event Hub Trigger.
  */
 class Function {
-    // Event Hub Trigger 
-    @FunctionName("structuralvalidator001")
-    def run(
-        @EventHubTrigger(
-            name = "msg",
-            eventHubName = "eventhub002",
-            connection = "EventHubConnectionString") 
-            message: String,
-            context: ExecutionContext): Unit = {
-
-            context.getLogger.info("Event hub triggered a request")
-            // context.getLogger.info("message: -> " + message)
+  // Event Hub Trigger
+  @FunctionName("structuralvalidator001")
+  def run(
+           @EventHubTrigger(
+             name = "msg",
+             eventHubName = "%EventHubReceiveName%",
+             connection = "EventHubConnectionString")
+           message: String,
+           context: ExecutionContext): Unit = {
 
-            val evHubNameOk = System.getenv("EventHubSendOkName")
-            val evHubNameErrs = System.getenv("EventHubSendErrsName")
-            val evHubConnStr = System.getenv("EventHubConnectionString")
+    context.getLogger.info("Event hub triggered a request")
+    // context.getLogger.info("message: -> " + message)
 
+    val evHubNameOk = System.getenv("EventHubSendOkName")
+    val evHubNameErrs = System.getenv("EventHubSendErrsName")
+    val evHubConnStr = System.getenv("EventHubConnectionString")
 
-            val hl7Messages = JsonUtil.fromJson[List[HL7Message]](message)
-            // context.getLogger.info(s"hl7Messages: --> ${hl7Messages}")
 
+    val elem = JsonParser.parseString(message.toString())
+    // context.getLogger.info("Json Array size:" + elem.getAsJsonArray)
+    val elemArray = elem.getAsJsonArray()
+    // var msgObj = elem.getAsJsonArray.get(0).getAsJsonObject.get("content").getAsString
+    // var metadataObj = elem.getAsJsonArray.get(0).getAsJsonObject.get("metadata").toString
+    // context.getLogger.info(s"metadataObj : ${metadataObj}")
 
-            // for each message received from event hub:
-            for (hl7Message <- hl7Messages) {
+    // for each message received from event hub:
+    var itor: util.Iterator[JsonElement] = elemArray.iterator()
+    while (itor.hasNext) {
```

Comment:
Instead of using an iterator - can you try this:
//Add Import statement:
    import scala.collection.JavaConversions._

//switch while (itor.hasNext) for a for Loop
    for( msgElem <- elemArray) {